Pinta.Core/Extensions/Cairo/CairoExtensions.ColorType.cs

```diff
@@ -58,25 +58,30 @@ public string ToHex (bool addAlpha = true)
 	/// <returns>Resulting color. If null, the method could not parse it.</returns>
 	public static Color? FromHex (string hex)
 	{
-		if (hex.StartsWith ("#"))
-			hex = hex.Remove (0, 1);
+		string hashStripped =
+			hex.StartsWith ('#')
+			? hex[1..]
+			: hex;
 
 		// handle shorthand hex
-		if (hex.Length == 3)
-			hex = $"{hex[0]}{hex[0]}{hex[1]}{hex[1]}{hex[2]}{hex[2]}";
-		if (hex.Length == 4)
-			hex = $"{hex[0]}{hex[0]}{hex[1]}{hex[1]}{hex[2]}{hex[2]}{hex[3]}{hex[3]}";
+		string lengthAdjusted = hashStripped.Length switch {
```

Comment:
Perhaps this first section could be split out into a helper method for normalizing the hex string? I feel like this got less readable, especially the shorthand expansion (which maybe would be more clear to read as a loop with a string builder?)

Reply:
👍 Looks good - thanks!